src/services/UserService.ts

```diff
@@ -21,4 +21,27 @@ export default class UserService {
     users.push(newUser);
     return newUser;
   }
+
+  async updateUser(userId: string, user: User): Promise<User> {
+    const targetUser = await users.find(dbUser => dbUser.id === userId);
+    if (!targetUser) {
+      throw new HttpException('User not found', HttpStatus.BAD_REQUEST);
+    }
+
+    const usernameCheck = await users.find(dbUser => dbUser.username === user.username && dbUser.id !== userId);
+    if (usernameCheck) {
+      throw new HttpException('Username already exists', HttpStatus.NOT_FOUND);
+    }
+    targetUser.username = user.username;
```

Comment:
Just think about the user with lots of other props (name, address, cell number, .....)